### PR TITLE
Fixes #2424 output mapping serialization

### LINKED ISSUE #2424: OutputMapping should be exported to MoBi?
I think this belongs to the simulation transfer between PKSim and MoBi 
@rwmcintosh @PavelBal @Yuri05 @georgeDaskalakis 

or is there a reason why this was not done?

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: tests/PKSim.Tests/Core/DomainHelperForSpecs.cs [hunk 1]
```diff
@@ -291,7 +291,7 @@ public static SensitivityParameter SensitivityParameter(string name = "Sensitivi
 
       public static DataRepository IndividualSimulationDataRepositoryFor(string simulationName)
       {
-         var simulationResults = new DataRepository("Results");
+         var simulationResults = new DataRepository();
```
Comment (msevestre) [inline tests/PKSim.Tests/Core/DomainHelperForSpecs.cs:294]: Id was fixed here. Letting the code create the id automatically

File: src/PKSim.Core/Services/ParameterTask.cs [hunk 2]
```diff
@@ -452,12 +452,12 @@ public ICommand UpdateDistributedTableFormula(IParameter tableParameter, IDistri
...
       }
 
-      public ICommand SetParameterValue(IParameter parameter, double value, ISimulation simulation)
+      public ICommand SetParameterValue(IParameter parameter, double value, IModelCoreSimulation simulation)
       {
          return SetParameterValue(parameter, value);
       }
 
-      public ICommand UpdateParameterValueOrigin(IParameter parameter, ValueOrigin valueOrigin, ISimulation simulation)
+      public ICommand UpdateParameterValueOrigin(IParameter parameter, ValueOrigin valueOrigin, IModelCoreSimulation simulation)
```
Comment (msevestre) [inline src/PKSim.Core/Services/ParameterTask.cs:460]: change related to the change of interface

File: src/PKSim.Infrastructure/Services/MoBiExportTask.cs [hunk 3]
```diff
@@ -114,6 +114,7 @@ private void exportSimulationToFile(Simulation simulation, string moBiFile)
          var simulationTransfer = new SimulationTransfer
          {
             Simulation = moBiSimulation,
+            OutputMappings = simulation.OutputMappings,
```
Comment (msevestre) [inline src/PKSim.Infrastructure/Services/MoBiExportTask.cs:117]: saving the actual output mapping when exporting

File: src/PKSim.Infrastructure/Services/SimulationTransferLoader.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ public SimulationTransfer Load(string pkmlFileFullPath)
       {
          var project = _projectRetriever.CurrentProject;
 
-         //use new ObjectBaseRepository here as the resulting simulation will be registered later on when added to the project
+         //use new WithIdRepository here as the resulting simulation will be registered later on when added to the project
          var simulationTransfer = _simulationPersister.Load(pkmlFileFullPath, new WithIdRepository());
```
Comment (msevestre) [inline src/PKSim.Infrastructure/Services/SimulationTransferLoader.cs:31]: Interestingly, we do not load obs data when loading from MoBi (this is not really a use case that is done often). So I am not going to tackle this for now

**Review by rwmcintosh — APPROVED** [review]

**Pull request merged**